python/ray/data/tests/test_actor_pool_map_operator.py

```diff
@@ -537,6 +618,8 @@ def __call__(self, x):
 
     from ray.exceptions import GetTimeoutError
 
+    ray.shutdown()
+    ray.init()
```

Comment:
why this is change needed?

Reply:
Without this change, cluster started by TestActorPool was still running.

Reply:
@srinathk10 did you run into this issue locally or in CI?

In general, I'd be cautious to explicitly call shutdown or init in unit tests because 1) it can cause issues where unit tests depend on their order in the module (i.e., tests aren't independent) and 2) calling init is relatively slow, and it's one of the largest sources of slowness in our test suite

Reply:
I was running into this locally once I added test_actor_pool_fault_tolerance_e2e. Looks like some side effect between tests in this pytest suite.


    def test_start_actor_timeout(ray_start_regular_shared, restore_data_context):
        """Tests that ActorPoolMapOperator raises an exception on
        timeout while waiting for actors."""

        class UDFClass:
            def __call__(self, x):
                return x

        from ray.exceptions import GetTimeoutError

>       ray.init()

test_actor_pool_map_operator.py:628:
>               raise RuntimeError(
                    "Maybe you called ray.init twice by accident? "
                    "This error can be suppressed by passing in "
                    "'ignore_reinit_error=True' or by calling "
                    "'ray.shutdown()' prior to 'ray.init()'."
                )
E               RuntimeError: Maybe you called ray.init twice by accident? This error can be suppressed by passing in 'ignore_reinit_error=True' or by calling 'ray.shutdown()' prior to 'ray.init()'.

Reply:
I think if you change `ray_start_regular_shared` to `ray_start_regular` and remove `ray.shutdown()` in the new test, it should work

Reply:
Actually removing ray.shutdown() and ray.init() got rid of this issue.